File: 1-Iniciante/Ex1435.java
import java.util.Scanner;

public class Ex1435 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int n = sc.nextInt();
            int temp = n;
            if(n == 0) break; // condicao de parada
            
            int[][] mat = new int[n][n];

            if(n < 3){
                preencheMat(mat, 0, n - 1, 1);
            }
            else{
                int ini = 0;
                int fim = n - 1;
                int val = 1;

                while(n >= 2){
                    preencheMat(mat, ini, fim, val);
                    ini++;
                    fim--;
                    val++;
                    n--;
                }
            }
            printMat(mat, temp);
        }

    }

    public static void preencheMat(int[][] mat, int inicio, int fim, int valor){
        for(int i = inicio; i <= fim; i++){
            for(int j = inicio; j <= fim; j++){
                mat[i][j] = valor;
            }
        }
    }
    
    public static void printMat(int[][] mat, int ord){
        for(int i = 0; i < ord; i++){
            for(int j = 0; j < ord; j++){
                if(j == 0) System.out.printf("%3d", mat[i][j]);
                else System.out.printf("%4d", mat[i][j]);
            }
            System.out.println();
        }
        System.out.println();
    }
}

File: 1-Iniciante/ex1435.py
def preencheMat(matriz, inicio, fim, valor): #funcao que vai preencher a matriz
    for i in range(inicio, fim):             #sempre de forma quadrada           
        for j in range(inicio, fim):         #uma matriz quadrada de ordem maior que 2 possui outras matrizes quadradas dentro dela
            mat[i][j] = valor                #baseando-se nisso essa solução foi desenvolvida   
                                             #codigo comentado para posterior compreenssão   
#entrada                                     #depure o codigo para entender melhor   
while True:
    mat = []
    n = int(input())

    if n == 0:
        break
    
    for i in range(0, n):  #criacao da matriz
        mat = mat + [[1] * n]

    fim = n - 1
    valor = 2
    if n > 2:
        for i in range(1, n - 1): #preenchendo a matriz
            preencheMat(mat, i, fim, valor)
            valor += 1
            fim -= 1

    for i in range(0, n): #saida
        for j in range(0, n):
            if j < n - 1:
                print('{:>3}'.format(mat[i][j]), end=' ')
            else:
                print('{:>3}'.format(mat[i][j]), end='')
        print()
    print()
